core/arch/arm/include/kernel/spinlock.h

```diff
@@ -69,7 +69,7 @@ static inline unsigned int cpu_spin_trylock(unsigned int *lock)
 
 	assert(thread_irq_disabled());
 	locked = __cpu_spin_trylock(lock);
-	if (locked)
+	if (!locked)
```

Comment:
This is confusing. Perhaps `cpu_spin_trylock()` should return a `bool` instead to align better with `mutex_trylock()`.
Then I'm also wondering what is really returned by the AArch64 version of `__cpu_spin_trylock()`.